src/templates/header.js

```diff
@@ -11,20 +12,91 @@ const links = [
   {title: 'Contact', link: '/contact'},
 ]
 
-const Header = ({ siteTitle }) => {
-  return (
-  <header>
-    <div>
-      <Nav 
-        logo={Logo({title: siteTitle})}
-        links={links}
-        GatsbyLink={Link}
-      />
-    </div>
-  </header>
-)
+class Header extends Component {
+
+  componentDidMount() {
+    if(typeof window !== 'undefined') {
+      let userDetails = {}
+      let SITE_NAME = window.location.origin
+      document.getElementById('contact-form') &&
+      document
+        .getElementById('contact-form')
+        .addEventListener('submit', (e => {
+          e.preventDefault()
+          const idea = document.getElementById('idea').value
+          const email = document.getElementById('email').value
+          document.getElementById('contact-submit').disabled = true
+          document.getElementById('idea').value = ""
+          document.getElementById('email').value = ""
+  
+          const formSubmitData = {
+            email,
+            idea,
+            userDetails,
+          }
+
+          const http = new XMLHttpRequest()
+          http.open('POST', hooksUrl)
+          http.onreadystatechange = function() {
+            if(http.readyState === XMLHttpRequest.DONE && http.status === 200) {
+              localStorage.setItem('userDetails', JSON.stringify({}))
+              document.getElementById('contact-submit').disabled = false
+              navigate('/thank-you')
+            }
+          }
```

Comment:
In case if the request fails, you're not enabling the button. Also, you are not handling the error case.